packages/@sanity/base/src/search/weighted/createWeightedSearch.js

```diff
@@ -7,9 +7,16 @@ import {tokenize} from '../common/tokenize'
 
 const combinePaths = flow([flatten, union, compact])
 
+const isArrayPath = path => path.endsWith('[]')
+
+// "some[].thing[]" => "some[].thing"
+const joinedPathRemoveArray = path => path.substr(0, path.length - 2)
+
 const toGroqParams = terms =>
   terms.reduce((acc, term, i) => {
     acc[`t${i}`] = `${term}*` // "t" is short for term
```

Comment:
Couldn't we just append the `*` later? Why does it need to be included in the terms like this?